### ISSUE TITLE
V4: FlatList/ScrollView className bug - functional components cannot be given refs

### ISSUE THREAD

**Issue by slvstr-dev** [marklawlor/nativewind#672]
**Describe the bug**
I trying to add TW classes to `FlatList` and `ScrollView`-components through the `className` prop and I'm running into the issue below:

```
Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?

Check the render method of `NativeViewGestureHandler`.
    in CSSInterop.ScrollView (created by NativeViewGestureHandler)
    in NativeViewGestureHandler (created by ScrollView)
    in ScrollView
    in Unknown (created by VirtualizedList)
    in VirtualizedListContextProvider (created by VirtualizedList)
    in VirtualizedList
    in render (created by FlatList)
    in FlatList
```

**To Reproduce**
1. Install `"expo": "49.0.20"` (I'm also using `expo-router`), and `"nativewind": "4.0.16" (also ran into this issue on `"4.0.13"`).
2. Render a `FlatList`-component (from `react-native` or `react-native-gesture-handler`):
	- E.g. `<FlatList data={list} className="p-10" renderItem={({ item }) => <Text user={item} />} />`
3. Add TW-classes through the `className` prop.
4. Check console warnings.

**Expected behavior**
I expect to be able to add TW classes to the `className` prop on a `FlatList` and `ScrollView` without a ref error.

**Comment by slvstr-dev** [comment]
Just noticed that `TextInput` from 'react-native-gesture-handler' also has this issue. `TextInput` and `ScrollView` from `react-native` don't have this issue. `FlatList`, as described above, has this issue on both packages.

**Comment by brerdem** [comment]
Did you use `remapProps`?

**Comment by slvstr-dev** [comment]
@brerdem no, but I shouldn't need to do that, right?

[Docs](https://www.nativewind.dev/v4/overview/#in-action):
````
import { CustomText } from "third-party-text-component";

export function BoldText(props) {
  // You just need to write `className="<your styles>"`
  return <CustomText className="text-bold" {...props} />;
}
```

**Comment by trentcharlie** [comment]
Did you find any solution or workaround for this issue?

**Comment by trentcharlie** [comment]
@marklawlor I noticed that making the following change in `react-native-css-interop/src/runtime/components/rendering.native.ts ` fixes this error but I am not sure if that'd have any side effects.
![Uploading Screenshot 2023-12-15 at 12.07.23.png…]()

**Comment by trentcharlie** [comment]
@marklawlor I noticed that making the following change in `react-native-css-interop/src/runtime/components/rendering.native.ts ` fixes this error but I am not sure if that'd have any side effects.
<img width="1126" alt="Screenshot 2023-12-15 at 12 07 23" src="https://github.com/marklawlor/nativewind/assets/127359753/ed2cbfd4-6707-4ac2-bbc7-3be1ea35d605">

**Comment by trentcharlie** [comment]
> Just noticed that `TextInput` from 'react-native-gesture-handler' also has this issue. `TextInput` and `ScrollView` from `react-native` don't have this issue. `FlatList`, as described above, has this issue on both packages.

I am not sure if everyone experiences it this way but for me the app crashes with the above warning in console logs and the following in the development build.

![IMG_0108](https://github.com/marklawlor/nativewind/assets/127359753/441b0524-84f4-460e-941c-2a50c205a2ff)

**Comment by trentcharlie** [comment]
From the little I can infer, looks like it could be that `delete props.ref` cause a null to be passed down somewhere where a ref is expected

**Comment by fran-cadenas-fu** [comment]
I'm having the same problem with the TextInput.

**Comment by trentcharlie** [comment]
This issue seems to have been addressed in `4.0.22`

**Comment by marklawlor** [comment]
This *might* be fixed in `4.0.22`. I'm going to close this issue for now, but if you still have issues please comment
